src/Orleans.Runtime/MembershipService/SiloStatusListenerManager.cs

```diff
@@ -146,7 +146,8 @@ void ILifecycleParticipant<ISiloLifecycle>.Participate(ISiloLifecycle lifecycle)
         {
             var tasks = new List<Task>();
 
-            lifecycle.Subscribe(nameof(SiloStatusListenerManager), ServiceLifecycleStage.RuntimeInitialize, OnStart, OnStop);
+            lifecycle.Subscribe(nameof(SiloStatusListenerManager), ServiceLifecycleStage.AfterRuntimeGrainServices, OnStart, _ => Task.CompletedTask);
+            lifecycle.Subscribe(nameof(SiloStatusListenerManager), ServiceLifecycleStage.RuntimeInitialize, _ => Task.CompletedTask, OnStop);
```

Comment:
Status notifications were being sent out before cleanup was performed & this was causing subscribers (eg, `TypeManager`) to try to refresh type information for the superseded silo